src/apiserver/api_service_impl.cc

```diff
@@ -0,0 +1,301 @@
+        }
+    }
+    return true;
+}
+bool APIServiceImpl::AppendJsonValue(const butil::rapidjson::Value& v, hybridse::sdk::DataType type,
+                                     std::shared_ptr<fedb::sdk::SQLRequestRow> row) {
+    switch (type) {
+        case hybridse::sdk::kTypeBool:
+            row->AppendBool(v.GetBool());
+            break;
+        case hybridse::sdk::kTypeInt16:
+            row->AppendInt16(v.GetInt());  // TODO(hw): cast
+            break;
+        case hybridse::sdk::kTypeInt32:
+            row->AppendInt32(v.GetInt());
+            break;
+        case hybridse::sdk::kTypeInt64:
+            row->AppendInt64(v.GetInt64());
+            break;
+        case hybridse::sdk::kTypeFloat:
+            row->AppendFloat(v.GetDouble());  // TODO(hw): cast
+            break;
+        case hybridse::sdk::kTypeDouble:
+            row->AppendDouble(v.GetDouble());
+            break;
+        case hybridse::sdk::kTypeString:
+            row->Init(v.GetStringLength());  // TODO(hw): cast
+            row->AppendString(v.GetString());
+            break;
+        case hybridse::sdk::kTypeDate:
+            LOG(INFO) << v.GetString() << " convert not supported";
+            row->AppendDate(0);
+            break;
+        case hybridse::sdk::kTypeTimestamp:
+            row->AppendTimestamp(v.GetInt64());
+            break;
+        default:
+            return false;
+    }
+    return true;
+}
+
+void APIServiceImpl::RegisterPut() {
+    provider_.put("/db/:db_name/table/:table_name", [this](const InterfaceProvider::Params& param,
+                                                           const butil::IOBuf& req_body, JsonWriter& writer) {
+        auto err = GeneralError();
+        auto db_it = param.find("db_name");
+        auto table_it = param.find("table_name");
+        if (db_it == param.end() || table_it == param.end()) {
+            writer& err.Set("Invalid path");
+            return;
+        }
+        auto db = db_it->second;
+        auto table = table_it->second;
+
+        // json2doc, then generate an insert sql
+        Document document;
+        if (document.Parse(req_body.to_string().c_str()).HasParseError()) {
+            writer& err.Set("Json parse failed");
+            return;
+        }
+
+        const auto& value = document["value"];
+        // value should be array, and multi put is not supported now
+        if (!value.IsArray() || value.Empty() || value.Size() > 1) {
+            writer& err.Set("Invalid value in body");
+            return;
+        }
+        const auto& arr = value[0];
+        StringBuffer buffer;
+        Writer<StringBuffer> sql_writer(buffer);
+        arr.Accept(sql_writer);
+        std::string line(buffer.GetString());
+        std::string insert_sql = "insert into " + table + " values(" + line.substr(1, line.length() - 2) + ");";
+
+        hybridse::sdk::Status status;
+        auto ok = sql_router_->ExecuteInsert(db, insert_sql, &status);
```

Comment:
It's better to use `PreparedStatement` or Apiserver will parse sql for every request.

Reply:
Just take a look at https://github.com/4paradigm/fedb/blob/main/src/sdk/sql_cluster_router.h#L123

Reply:
`SQLInsertRow ` needs schema. For example, we get '1' in json, we can't figure out what is it, it's int or int64 ...? Maybe we should add one method to get table schema later.

Reply:
fixed